### PR TITLE
add java classes main task

### PR DESCRIPTION
**Pull request by PavelYudzin**

### CODE REVIEW

File: java/src/main/java/com/company/automation/classes/main_task/Faculty.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    COMPUTER_AIDED_DESIGN("FACULTY OF COMPUTER-AIDED DESIGN"),
+    INFORMATION_TECHNOLOGIES("FACULTY OF INFORMATION TECHNOLOGIES AND CONTROL"),
+    RADIO_ENGINEERING("FACULTY OF RADIO ENGINEERING AND ELECTRONICS"),
+    COMPUTER_SYSTEMS("FACULTY OF COMPUTER SYSTEMS AND NETWORKS"),
+    INFO_COMMUNICATION("FACULTY OF INFORMATION COMMUNICATION"),
+    ENGINEERING("FACULTY OF ENGINEERING AND ECONOMICS");
+
+    private final String name;
+
+    Faculty(String name) {
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/Faculty.java]: add private identifier

File: java/src/main/java/com/company/automation/classes/main_task/Student.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    public Student(int id, String firstName, String secondName, String additionalName, int year, int month, int day, String address,
+                   String phoneNumber) {
+        this.id = id;
+        this.firstName = firstName;
+        this.secondName = secondName;
+        this.additionalName = additionalName;
+        this.birthDate = LocalDate.of(year, month, day);
+        this.address = address;
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/Student.java]: replace like 
public Student(...) {
        this(...);
        this.address = address;

File: java/src/main/java/com/company/automation/classes/main_task/Student.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    public Student(int id, String firstName, String secondName, String additionalName, int year, int month, int day, String address,
+                   String phoneNumber, Course course, int group, Faculty faculty) {
+        this.id = id;
+        this.firstName = firstName;
+        this.secondName = secondName;
+        this.additionalName = additionalName;
+        this.birthDate = LocalDate.of(year, month, day);
+        this.address = address;
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/Student.java]: The same as was mentioned before

File: java/src/main/java/com/company/automation/classes/main_task/Student.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    private String secondName;
+    private String additionalName;
+    private LocalDate birthDate;
+    private String address;
+    private String phoneNumber = "";
+    private int group;
+    private Course course;
+    private Faculty faculty;
+
+    public Student(int id, String firstName, String secondName, String additionalName, int year, int month, int day, String address) {
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/Student.java]: You can use LocalDate as a constructor parameter and create the Student like:
Student student07 = new Student(64, "Lisa", "Simpson", "", LocalDate.of(1982, 3, 17), "USA, Springfield", "555-29-14", Course.SECOND, 122199, Faculty.INFO_COMMUNICATION);

File: java/src/main/java/com/company/automation/classes/main_task/Student.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+                ", phoneNumber='" + phoneNumber + '\'' +
+                ", course=" + course +
+                ", group=" + group +
+                ", faculty=" + faculty +
+                '}';
+    }
+
+    public static Student[] getArrayOfStudents(Student... student) {
+        return student;
+    }
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/Student.java]: Move it from this class. This method doesn't belong to the logic of this class

File: java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    public static StudentLogic getStudentLogic() {
+        return studentLogic;
+    }
+
+    public String getStudentFullInfo(Student student) {
+        return "Id: " + student.getId() + ". " + student.getFullName() + ". Birth date: " + student.getBirthDate() +
+                "\nAddress: " + student.getAddress() + "\nPhone number: " + student.getPhoneNumber() + "\n" +
+                student.getFaculty().getName() + "\n" + student.getCourse().getNumber() + " course. Group: " + student.getGroup() + "\n";
+    }
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java]: move it to the Student class

File: java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    public Student[] getStudentsOfGroup(Student[] students, int group) {
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java]: I think it should have a name like getSortedStudentsOfGroup

File: java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+public class StudentLogic {
+    private final static StudentLogic studentLogic = new StudentLogic();
+
+    private StudentLogic() {
+    }
+
+    public static StudentLogic getStudentLogic() {
+        return studentLogic;
+    }
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java]: I think that this way of realizing of singleton will be better =):

public class StudentLogic {
    private static StudentLogic studentLogic;

    private StudentLogic() {
    }

    public static StudentLogic getStudentLogic() {
        if (studentLogic == null) {
            studentLogic = new StudentLogic();
        }
        return studentLogic;
    }
}

File: java/src/main/java/com/company/automation/classes/main_task/StudentView.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+public class StudentView {
+    private final StudentLogic studentLogic = StudentLogic.getStudentLogic();
+    private static final StudentView studentView = new StudentView();
+
+    private StudentView() {
+    }
+
+    public static StudentView getStudentView() {
+        return studentView;
+    }
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/StudentView.java]: The same as mentioned before about singleton

File: java/src/main/java/com/company/automation/classes/main_task/StudentView.java [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+
+    public void printStudentsOfGroup(Student[] students, int group) {
+        int i = 1;
+        Student[] studentsOfGroup = studentLogic.getStudentsOfGroup(students, group);
+        System.out.println("Group " + group + ":");
+        if (studentsOfGroup.length != 0) {
+            for (Student item : studentsOfGroup) {
```
Comment (VitaliLebedzeu) [inline java/src/main/java/com/company/automation/classes/main_task/StudentView.java]: use the common `for (;;) {}` loop please )

File: java/src/main/java/com/company/automation/classes/main_task/Faculty.java (hunk 1 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/Faculty.java]: Fixed
Modifier 'private' is redundant for enum constructors

File: java/src/main/java/com/company/automation/classes/main_task/Student.java (hunk 2 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/Student.java]: Fixed

File: java/src/main/java/com/company/automation/classes/main_task/Student.java (hunk 3 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/Student.java]: Fixed

File: java/src/main/java/com/company/automation/classes/main_task/Student.java (hunk 4 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/Student.java]: Fixed

File: java/src/main/java/com/company/automation/classes/main_task/Student.java (hunk 5 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/Student.java]: Fixed

File: java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java (hunk 8 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java]: Fixed

File: java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java (hunk 6 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java]: fixed

File: java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java (hunk 7 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/StudentLogic.java]: fixed

File: java/src/main/java/com/company/automation/classes/main_task/StudentView.java (hunk 9 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/StudentView.java]: Fixed

File: java/src/main/java/com/company/automation/classes/main_task/StudentView.java (hunk 10 above)
Comment (PavelYudzin) [reply java/src/main/java/com/company/automation/classes/main_task/StudentView.java]: fixed

**Review by VitaliLebedzeu — APPROVED** [review]

**Pull request merged**